mapping/admin_set.rb

```diff
@@ -1,6 +1,29 @@
 # frozen_string_literal: true
 module Mapping
   class AdminSet
+    require 'csv'
+    require 'yaml'
+
     extend Extensions::BasicValueHandler
+
+    ##
+    # Convert data in a csv file to hash in YAML format
+    #
+    # @param[String] value - the path to the csv file
+    # return - [Hash] of YAML
+    def csv_to_yaml(value)
+      data = CSV.read (value, :headers => true).map(&:to_hash)
+      data.to_yaml
+    end
+
+    ##
+    # Locate adminset with the passing owning_collection handle
+    #
+    # @param[String] value - handle of owning collection
+    # return - admin_set 
+    def lookup_admin_set(value)
+      lookup = csv_to_yaml('../../lookup/collectionlist.csv')
+      lookup.select { |l| l[:collection_handle] == value }
```

Comment:
this gives you rows where the `:collection_handle` matches the value that's passed into `lookup_admin_set`, but you want to return the admin_set_id from this row. Per our discussion about adding an `admin_set_id` column to the CSV, you want to return the value for the one row.

```
# find the first row where collection_handle matches
row = lookup.find { |l| l[:collection_handle] == value } 
row[:admin_set_id]
```